spec/spec_helper.rb

```diff
@@ -39,11 +39,15 @@ def migrate(renames = {})
   end
 
   def nuke_model_class(klass)
-    ActiveSupport::DescendantsTracker.instance_eval do
-      direct_descendants = class_variable_get('@@direct_descendants')
-      direct_descendants[ActiveRecord::Base] = direct_descendants[ActiveRecord::Base].to_a.reject { |descendant| descendant == klass }
-      if defined?(ApplicationRecord)
-        direct_descendants[ApplicationRecord] = direct_descendants[ApplicationRecord].to_a.reject { |descendant| descendant == klass }
+    if ActiveSupport.version >= Gem::Version.new('7.1.0') || (ActiveSupport.version >= Gem::Version.new('7.0.0') && Class.method_defined?(:subclasses))
+      ActiveSupport::DescendantsTracker.clear([klass])
```

Comment:
📓 With Ruby 3.1 the `subclasses` method introduced to be able to track descendants, and Rails 7+ started to take advantage of this, allowing for a sleek new way to clear a class from the descendant tracking. The internals now just track and "exclude" list and this `clear` method appends to that list.